### PR TITLE
Alerting: Consolidate mute timings selector logic

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**

Similar to https://github.com/grafana/grafana/pull/91690 but for mute timings - consolidates the two places that were providing mute timings selector functionality

**Why do we need this feature?**

🧹 and also making sure that the right code flows through the K8S API when enabled

### CODE REVIEW

File: public/app/features/alerting/unified/components/alertmanager-entities/MuteTimingsSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+});
+
+/** Provides a MultiSelect with available mute timings for the given alertmanager */
+const MuteTimingsSelector = ({
+  alertmanager,
+  selectProps,
+}: BaseAlertmanagerArgs & { selectProps: MultiSelectCommonProps<string> }) => {
+  const { data } = useSelectableMuteTimings({ alertmanager });
+
+  const muteTimingOptions = useMemo(() => {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/alertmanager-entities/MuteTimingsSelector.tsx:23]: is it necessary this useMemo?

Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/alertmanager-entities/MuteTimingsSelector.tsx:23]: is it necessary this useMemo?

File: public/app/features/alerting/unified/components/mute-timings/useMuteTimings.tsx [hunk 2]
```diff
@@ -294,3 +297,19 @@ export const useValidateMuteTiming = ({ alertmanager }: BaseAlertmanagerArgs) =>
...
 };
+
+/**
+ * @deprecated This will be deprecated by the K8S API.
+ * Once that is enabled by default, this should be removed and the `getMuteTimingList` should be removed
+ */
+export const useSelectableMuteTimings = ({ alertmanager }: BaseAlertmanagerArgs) => {
+  const useK8sApi = shouldUseK8sApi(alertmanager);
+
+  const fetchGrafanaMuteTimings = useGetMuteTimingListQuery(undefined, {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/mute-timings/useMuteTimings.tsx:308]: this naming is confusing for me, as I see that when we are using k8s api we use the naming 'Grafana' , and when we use alertmanager api we use the naming alertmanager. 
I would expect `fetchGrafanaMuteTimings` to be something like `fetchAlertManagerMuteTimings` ? or is something I'm missing here?

File: public/app/features/alerting/unified/components/mute-timings/useMuteTimings.tsx [hunk 3]
```diff
@@ -294,3 +297,19 @@ export const useValidateMuteTiming = ({ alertmanager }: BaseAlertmanagerArgs) =>
...
+    skip: useK8sApi,
+  });
+
+  const fetchK8sMuteTimings = useMuteTimings({ alertmanager, skip: !useK8sApi });
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/mute-timings/useMuteTimings.tsx:312]: and in here I would expect using `fetchGrafanaMuteTimings` ?

File: public/app/features/alerting/unified/components/alertmanager-entities/MuteTimingsSelector.tsx (hunk 1 above)
Comment (tomratcliffe) [reply public/app/features/alerting/unified/components/alertmanager-entities/MuteTimingsSelector.tsx]: Nope! Removed, thanks

File: public/app/features/alerting/unified/components/mute-timings/useMuteTimings.tsx (hunk 2 above)
Comment (tomratcliffe) [reply public/app/features/alerting/unified/components/mute-timings/useMuteTimings.tsx]: I've changed the naming and the logic - I think I was previously calling the wrong API if we hadn't enabled the k8s API - what do you think of it now?

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM! Great work!

**Pull request merged**